include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -609,12 +625,16 @@ class charge_distribution_surface<Lyt, false> : public Lyt
         for (const auto& it : strg->loc_pot)  // this for-loop checks if the "population stability" is fulfilled.
         {
             bool valid = (((strg->cell_charge[for_loop_counter] == sidb_charge_state::NEGATIVE) &&
-                           ((-it + strg->phys_params.mu) < physical_constants::POP_STABILITY_ERR)) ||
+                           ((units::energy::electron_volt_t(-it.value()) + strg->phys_params.mu).value() <
+                            physical_constants::POP_STABILITY_ERR)) ||
                           ((strg->cell_charge[for_loop_counter] == sidb_charge_state::POSITIVE) &&
-                           ((-it + strg->phys_params.mu_p) > -physical_constants::POP_STABILITY_ERR)) ||
+                           ((units::energy::electron_volt_t(-it.value()) + strg->phys_params.mu_p).value() >
+                            -physical_constants::POP_STABILITY_ERR)) ||
                           ((strg->cell_charge[for_loop_counter] == sidb_charge_state::NEUTRAL) &&
-                           ((-it + strg->phys_params.mu) > -physical_constants::POP_STABILITY_ERR) &&
-                           (-it + strg->phys_params.mu_p) < physical_constants::POP_STABILITY_ERR));
+                           ((units::energy::electron_volt_t(-it.value()) + strg->phys_params.mu).value() >
+                            -physical_constants::POP_STABILITY_ERR) &&
+                           (units::energy::electron_volt_t(-it.value()) + strg->phys_params.mu_p).value() <
+                               physical_constants::POP_STABILITY_ERR));
```

Comment:
Are the `.value()` calls needed here? Again, I'm not yet familiar with the library and this is an honest question.